flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_Engine.js

```diff
@@ -284,5 +233,63 @@ var A320_Neo_LowerECAM_Engine;
         }
     }
     A320_Neo_LowerECAM_Engine.EngineInfo = EngineInfo;
+    class EngineIgniter {
+        constructor(_ignValueText) {
+            this.currentState = A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.NONE;
+            this.titleUpdateRequired = false;
+            this._ignValueText = _ignValueText;
+            this._lastUsedIgniter = A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.NONE;
+        }
+        updateStatus(_engineStarting, _n2Igniting, _n2Percent) {
+            const targetState = this.getTargetState(_engineStarting, _n2Igniting, _n2Percent);
+            if (this.currentState != A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.NONE && targetState != A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.NONE) {
+                this._lastUsedIgniter = this.currentState;
+            } else {
+                if (targetState != this.currentState) {
+                    this.currentState = targetState;
+                    if (this._ignValueText != null) {
+                        this._ignValueText.textContent = this.getIGNStringFromState(targetState);
+                        this.titleUpdateRequired = true;
+                    }
+                } else {
+                    this.titleUpdateRequired = false;
+                }
+            }
+        }
+        getTargetState(_engineStarting, _n2Igniting, _n2Percent) {
+            let targetState = A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.NONE;
+            if (_engineStarting && _n2Igniting && _n2Percent > 18 && _n2Percent < 55) {
+                if (this._lastUsedIgniter === A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.NONE) {
+                    targetState = A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.A;
+                } else if (this._lastUsedIgniter === A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.A) {
+                    targetState = A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.B;
+                } else if (this._lastUsedIgniter === A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.B) {
+                    targetState = A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.A;
+                }
+            }
+            return targetState;
+        }
```

Comment:
**suggestion**: Returning early is usually preferred.
**suggestion**: The returning of state `A` for two separate conditions can be merged.

```suggestion
        getTargetState(engineStarting, n2Igniting, n2Percent) {
            if (engineStarting && n2Igniting && n2Percent > 18 && n2Percent < 55) {
                if (this._lastUsedIgniter === A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.NONE ||
                    this._lastUsedIgniter === A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.B) {

                    return A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.A;
                } else if (this._lastUsedIgniter === A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.A) {
                    return A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.B;
                }
            }

            return A320_Neo_LowerECAM_Engine.Definitions.IGN_STATE.NONE;
        }
```

Reply:
I had thought it was easier to comprehend what was happening with the separation of each state into it's own conditional. Though looking at your suggestion, it keeps the code more concise and ultimately doesn't add much overhead to comprehension. And of course is better for extensibility and maintainability. Done.